src/backends/caffe2/netdef_backend.cc

```diff
@@ -453,6 +453,36 @@ NetDefBackend::Context::ReadFixedSizedOutputTensor(
     return Status(RequestStatusCode::INTERNAL, err.Message());
   }
 
+  // verify shape of output matches shape from model config
+  const int batch_offset = ((max_batch_size_ == NO_BATCHING) ? 0 : 1);
+
+  for (int i = 0; i < dims.size(); i++) {
+    if (dims[i] != -1) {
+      if (dims[i] != content_shape[i + batch_offset]) {
```

Comment:
Use utility methods from model_config.h to create string representation of dims.